internal/pkg/stream/stream_test.go

```diff
@@ -136,4 +136,15 @@ func TestStream(t *testing.T) {
 		require.Greater(t, streamer.fetchCount, 1, "expected more than one call to Fetch within timeout")
 		require.Greater(t, streamer.notifyCount, 1, "expected more than one call to Notify within timeout")
 	})
+
+	t.Run("nextFetchDate works correctly to grab times before the timeout.", func(t *testing.T) {
```

Comment:
I think we can now remove this test case since we can ensure that multiple calls to `Fetch` will double the interval on each call